### ISSUE TITLE
Explanation of config / extending logging

### ISSUE THREAD

**Issue by smoochy** [bmino/binance-triangle-arbitrage#22]
Hi,

I do have a few questions regarding the values in the config.

>      INVESTMENT: {
>         // Symbol which all triangle trades must start and end with
>         BASE: 'BTC',
>         MIN: 0.0,
>         MAX: 0.15,
>         STEP: 0.001
>     },

The MIN and MAX settings, are only regarding the base currency and the first trade from A (BTC in that case) to B?
That means, the bot is looking to use between 0 and 0.15BTC to buy any altcoin. Afterwards, those settings do not matter anymore, as it is tried to always trade all of B into C and all of C into A again, correct?

> // Parallel: all three legs of the triangle arbitrage are executed in parallel
> EXECUTION_STRATEGY: 'Parallel',

Does that mean, that it is tried to trade
A to B
B to C
C to A

at the very same time? I assume this would require me to have a balance of B and C already, before that trade is initiated?

---------------------------------

Could you extend the logging into the execution.log with additions like:

> Found profitable trade: BTC-ADA-BNB, expecting 2.00% profit
> Used 0.023BTC to trade into 40 ADA: successful
> Used 35 ADA to trade into 3 BNB: successful. 5 ADA could not be traded ({ORDERBOOK not sufficient or other reason named}), left to dust. Is tried to be picked up in next ADA trade.
> Used 3 BNB to trade into 0.0235 BTC: successful: 2.17% Profit.

This makes it much clearer, what happened in the background and makes the trades much more comprehensible.

In addition, could you convert the unix timestamp into a human readable one, like UTC time?

> {"level":"info","time":1549415784366,"msg":"Buying 4.42 EOSTUSD @ market price","v":1}

->

>{"level":"info","time":02/06/2019 @ 1:16am (UTC),"msg":"Buying 4.42 EOSTUSD @ market price","v":1}

Cheers.

**Comment by bmino** [comment]
Hey!

You are absolutely correct with your interpretation of the `INVESTMENT.*` configurations

You are also correct with your interpretation of the Parallel execution strategy that it requires you hold appropriate balances of everything in the whitelist.

I already have release 4.1.0 pending which defaults the execution strategy to Linear as this is in line with what the average user expects.

I'm with you on the logging syntax for better describing what is going on and what happened. It is pretty crude currently.

As for the timestamp formatting that's a little of a doozy as formatting timestamps hits performance. This could be provided as a configuration option for users who deem the tradeoff worthwhile?

**Comment by smoochy** [comment]
Thanks for your reply.

If I am correct with the investment settings, how can issues like described [here](https://github.com/bmino/binance-triangle-arbitrage/issues/20#issuecomment-461127654) happen in the first place?

Lets say I have the following settings:

>  INVESTMENT: {
>     // Symbol which all triangle trades must start and end with
>     BASE: 'BTC',
>     MIN: 0.0015,
>     MAX: 0.2,
>     STEP: 0.001
> },

So I am using at least 0.0015BTC and a maximum of 0.2BTC for Trading A into B.
If I then get an amount x of B, and want to trade it into C. Should it not be provided, that I have enough of B to actually be able to trade it into C? And the same thing for trading C into A again.

There is no use in starting the trade at all, if I end up being stuck with B or C, as their MIN_NOTIONAL is not being met.

Wouldn't it be possible to check if this is the case, before starting the trade initially?

> As for the timestamp formatting that's a little of a doozy as formatting timestamps hits performance. This could be provided as a configuration option for users who deem the tradeoff worthwhile?

Shouldn't the time that would take to convert it be negligible? As being like 1ms or so?

**Comment by bmino** [comment]
Checking the MIN_NOTIONAL filter could be a precursory check before engaging in a position, yeah. You still run the risk of the market changing in the half second or so it takes to execute the AB leg and you end up with slightly less of B then you expect.

I agree with you on the timestamp formatting and lent cautiously at the recommendation of the pino docs

**Comment by bmino** [comment]
The logging changes have been implemented!
Please open another issue for discussing and and tracking the implementation of a MIN_NOTIONAL precheck
